File: apk/src/main/java/uk/co/alt236/apk/repo/dex/AccessFlagsResolver.java
package uk.co.alt236.apk.repo.dex;

import com.android.dx.rop.code.AccessFlags;
import uk.co.alt236.apk.repo.dex.model.DexClass;
import uk.co.alt236.apk.repo.dex.model.DexMethod;

import javax.annotation.Nonnull;

public final class AccessFlagsResolver {

    public static String resolve(@Nonnull DexClass item) {
        final String result;
        if (item.isInnerClass()) {
            result = AccessFlags.innerClassString(item.getAccessFlags());
        } else {
            result = AccessFlags.classString(item.getAccessFlags());
        }

        return sanitise(result);
    }

    public static String resolve(@Nonnull DexMethod item) {
        final String result = AccessFlags.methodString(item.getAccessFlags());

        return sanitise(result);
    }

    private static String sanitise(final String flagString) {
        if ("0000".equals(flagString)) {
            return "none";
        } else {
            return flagString;
        }
    }
}

File: apk/src/main/java/uk/co/alt236/apk/repo/dex/AccessFlagsResolver.kt
package uk.co.alt236.apk.repo.dex

import com.android.dx.rop.code.AccessFlags
import uk.co.alt236.apk.repo.dex.model.DexClass
import uk.co.alt236.apk.repo.dex.model.DexMethod

object AccessFlagsResolver {

    @JvmStatic
    fun resolve(item: DexClass): String {
        val result = if (item.isInnerClass) {
            AccessFlags.innerClassString(item.accessFlags)
        } else {
            AccessFlags.classString(item.accessFlags)
        }

        return sanitise(result)
    }

    @JvmStatic
    fun resolve(item: DexMethod): String {
        val result = AccessFlags.methodString(item.accessFlags)
        return sanitise(result)
    }

    private fun sanitise(flagString: String): String {
        return if ("0000" == flagString) {
            "none"
        } else {
            flagString
        }
    }
}
